src/bsd.rs

```diff
@@ -0,0 +1,42 @@
+use std::fs::read_link;
+use std::path::PathBuf;
+use libc::*;
+use std::ptr;
+
+// http://stackoverflow.com/q/799679
+// http://stackoverflow.com/a/1024937
+pub fn get_executable_path() -> Option<PathBuf> {
+    // FreeBSD
+    let mib = [CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1];
+    let mut buf: Vec<u8> = Vec::with_capacity(1024);
+    let mut cb = 1024 as size_t;
+    let result = unsafe {
+        sysctl(mib.as_ptr(),
```

Comment:
My only concern at this point, is that on NetBSD & DragonflyBSD, we're calling this sysctl that only works on FreeBSD. It seems that could have unintended consequences. I suggest we use the [`cfg!()` macro ](https://doc.rust-lang.org/std/macro.cfg.html) to do platform detection. For example:

```
if cfg!(target_os = "freebsd") {
//Freebsd sysctl call
} else {
//fallback to readlink on /proc/curproc/{exe,file}
}
```